ARTICLE.md

```diff
@@ -1,59 +1,144 @@
+description: This guide walks you through the what and why of property-based testing, with practical use cases and examples along the way. 
+author: Fredrik Fornwall
+authorLink: https://dev.to/fornwall
+tags:
+  - testing
+  - tutorial
+  - beginners
+  - python
+---
 
-![Evolution of humans ending up with computer usage](https://img.freepik.com/free-vector/human-evolution-from-monkey-businessman-computer-user-cartoon-vector-characters_53562-7997.jpg?size=626&ext=jpg)
+<!-- TODO: INTRODUCTION SENTENCE 🙆‍♀️ -->
 
-## Example based testing
-Normally software testing is done through **example based testing**. A human writes one or several sample inputs to the function or system under test, runs the function or system, and then asserts on the result of that.
+Maybe you've written unit tests before, but this is the first time you've heard about property-based testing. Or maybe you've heard the term, but still don't really get what it's about. Either way, we've got you. 
 
-Let's start with a toy example - a python [bubble sort](https://en.wikipedia.org/wiki/Bubble_sort) function to sort a list of numbers (note: this is a toy example, use [list.sort()](https://docs.python.org/3/library/stdtypes.html#list.sort) in production):
+In this guide, we'll cover the fundamentals of property-based testing and how it differs from traditional example-based testing practices. We'll walk you through practical examples of properties and how to write tests that address them. Finally, we'll learn how to use property-based testing to find bugs in our code and what existing libraries are out there. 
+
+## What's in this guide
+
+- [Traditional unit tests based on examples](#traditional-unit-tests-based-on-examples)
+    - [Limitations of example-based testing](#limitations-of-example-based-testing)
+- [Introduction to property-based testing](#introduction-to-property-based-testing)
+    - [An example using Hypothesis](#an-example-using-hypothesis)
+    - [What can be a property?](#what-can-be-a-property)
+    - [How does property-based testing differ from example-based?](#how-does-property-based-testing-differ-from-example-based)
+- [Example properties and how to test for them](#example-properties-and-how-to-test-for-them)
+    - [Unexpected exceptions should never be thrown](#unexpected-exceptions-should-never-be-thrown)
+    - [Values shouldn't change after encoding and then decoding](#values-shouldnt-change-after-encoding-and-then-decoding)
+    - [A naive method should still give the same result](#a-naive-method-should-still-give-the-same-result)
+- [Finding bugs with property-based testing](#finding-bugs-with-property-based-testing)
+- [Available libraries](#available-libraries)
+- [Conclusion](#conclusion)
+
+⚠️ Prerequisites:
+- A general understanding of how software testing works.
+- (Optional) [Python 3+](https://www.python.org/downloads/)* if you want to follow along in your own IDE. 
```

Comment:
Yes!!! Done

Reply:
What I'm wondering though, is if we should change the name of the repo? 🤔 I'll still update the text because changing the link after the fact is easy, but just a thought.

Reply:
Yea, having the repo name match the article title is good!

`beginners-guide-to-property-based-testing` ?

Reply:
Done!